src/xrpld/app/paths/detail/StepChecks.h

```diff
@@ -52,6 +52,15 @@ checkFreeze(
         {
             return terNO_LINE;
         }
+        if (view.rules().enabled(featureDeepFreeze))
```

Comment:
Probably don't need to check if enabled since a trustline can't be deep frozen unless the feature is enabled. And perhaps can simplify freeze/deep-freeze:

    if (sle->getFlags() &
            (lsfHighFreeze | lsfLowFreeze | lsfHighDeepFreeze |
             lsfLowDeepFreeze))
            return terNO_LINE;